adr/ADR-49.md

```diff
@@ -149,19 +154,31 @@ to reset it. In the replicated case a purge will not replicate, a roll-up would
 everywhere so the only real option is to publish a negative value into the particular point in the tree structure where
 you want to reset.
 
-In this scenario users must take care to configure limits appropriately, at no point may a down Source miss messages,
-this means if a Source is down for a day, there must be history to cover that period. Each aggregation point must also
-be configured to be a Counter.
-
 #### Adding Sources
 
-Adding a source that already has values in it and that had limits applied will be problematic since we will not have the 
-history to fully recount the message in the target Stream. Additionally if a Source had many messages adding in the whole
-source might be computationally too expensive, especially when the history is not important.
+Adding a source that uses a subject transform to merge an upstream counter into an aggregate one is supported through a
+source-tracking `Nats-Counter-Sources` header, which is automatically populated and maintained by the server.
+
+The `Nats-Counter-Sources` keys each source based on the stream name & original pre-transformed subject and contains the most
+recently seen `"val"` for each source.
+
+When a message is received from a source, as determined by the presence of the `Nats-Stream-Source` header, the server will
+not use the sourced `Nats-Incr` value to update the counter verbatim, but will instead rewrite `Nats-Incr` to reflect the
+delta between the last seen `"val"` for that source and the new one.
+
+This way, it is possible to replay `Nats-Incr` headers and recount on the aggregate stream even if some messages from the
+source were lost, i.e. due to message retention policies, network outages etc, and still arrive at the correct total.
+
+This also means that adding a source that has a non-zero counter is possible and the `Nats-Incr` header will include that
+initial count, i.e. adding source counter with an existing value of 10 will result in the first sourced count having a 
+`Nats-Incr: 10`. 
+
+Note that once a source key is added to this map, it is not ever removed, so that if the stream source is removed and
+re-added later to the stream config, the server will be able to do the right thing and determine the difference during
+that time, instead of re-adding the whole value and potentially ending up with a miscount on the aggregated stream.
 
-Tooling might use a direct get to retrieve that value in the source, place that value into the target Stream and then 
-start the sourcing from seq+1, meaning we snapshot the Source value into the target and then only handle new counts.  
-This will scale well and avoid huge recounts, but, would require the target Stream to have a subject.
+These properties mean that aggregated counters will always be eventually consistent with their sources, even if the
+aggregating stream also has local additions or subtractions that were not themselves sourced.
```

Comment:
I think we can remove the `Client-Assisted` section below as it primarily exist to address issues this new header fixes.